spl-balance/src/instruction.rs

```diff
@@ -0,0 +1,658 @@
+use crate::option::COption;
+use std::convert::TryInto;
+use substreams::errors::Error;
+
+/// Instructions supported by the token program.
+#[repr(C)]
+#[derive(Clone, Debug, PartialEq)]
+pub enum TokenInstruction<'a> {
```

Comment:
Shrink this enum down to having just three fields: 

 - InitializeMint
 - InitializeMint2
 - NonMintInstruction

You will also be able to remove all C api code and a bunch of the map code